typed/typed.go

```diff
@@ -213,14 +221,27 @@ func merge(lhs, rhs *TypedValue, rule, postRule mergeRule) (*TypedValue, error)
 			errorf("expected objects of the same type, but got %v and %v", lhs.typeRef, rhs.typeRef)
 	}
 
-	mw := mergingWalker{
-		lhs:          &lhs.value,
-		rhs:          &rhs.value,
-		schema:       lhs.schema,
-		typeRef:      lhs.typeRef,
-		rule:         rule,
-		postItemHook: postRule,
-	}
+	mw := mwPool.Get().(*mergingWalker)
+	defer func() {
+		mw.lhs = nil
```

Comment:
Does it work if you do `*mw = mergingWalker{}`?

Reply:
No, I'm deliberately preserving two of the fields...